src/typeahead/index.js

```diff
@@ -238,6 +242,41 @@ var Typeahead = React.createClass({
         value={ this.state.selection }
       />
     );
+  },
+
+  _generateFilterFunction: function() {
+    var filterOptionProp = this.props.filterOption;
+    if (typeof filterOptionProp === 'function') {
+      return function(value, options) {
+        return options.filter(function(o) { return filterOptionProp(value, o); });
+      };
+    } else {
+      var mapper;
+      if (typeof filterOptionProp === 'string') {
+        mapper = function(o) { return o[filterOptionProp]; };
+      } else {
```

Comment:
Can this ever happen?  I thought we just checked if typeof filterOptionProp was 'function' up above?